cypress/e2e/DoenetML/tagSpecific/codeeditor.cy.js

```diff
@@ -1344,4 +1344,107 @@ describe("Code Editor Tag Tests", function () {
     );
     cy.get(cesc2("#/pv")).should("have.text", "value: Hello!\nSelam!\nKaixo!");
   });
+
+  it("undo prompts save", () => {
+    let doenetML = `
+    <text>a</text>
+    <codeEditor showResults />
+
+    <p><copy prop="value" target="_codeeditor1" /></p>
+    `;
+
+    cy.get("#testRunner_toggleControls").click();
+    cy.get("#testRunner_allowLocalState").click();
+    cy.wait(100);
+    cy.get("#testRunner_toggleControls").click();
+
+    cy.window().then(async (win) => {
+      win.postMessage(
+        {
+          doenetML,
+        },
+        "*",
+      );
+    });
+
+    cy.get(cesc("#\\/_text1")).should("contain.text", "a");
+
+    cy.window().then(async (win) => {
+      let stateVariables = await win.returnAllStateVariables1();
+
+      let viewerName =
+        stateVariables["/_codeeditor1"].activeChildren[0].componentName;
+      let updateAnchor = "#" + cesc2(viewerName) + "_updateButton";
+      let contentAnchor = "#" + cesc2(viewerName) + "_content";
+
+      cy.get(cesc2("#/_p1")).should("have.text", "");
+      cy.get(contentAnchor).should("have.text", "");
+
+      cy.log("type text in editor");
+      cy.get(cesc("#\\/_codeeditor1") + " .cm-content").type("<p>Hello!</p>", {
+        delay: 0,
+      });
+      cy.get(updateAnchor).click();
+
+      cy.get(cesc2("#/_p1")).should("have.text", "<p>Hello!</p>");
+      cy.get(contentAnchor).should("have.text", "Hello!");
+
+      cy.wait(1500); // wait for 1 second debounce
+
+      cy.reload();
+
+      cy.window().then(async (win) => {
+        win.postMessage(
+          {
+            doenetML,
+          },
+          "*",
+        );
+      });
+
+      cy.get(cesc2("#/_p1")).should("have.text", "<p>Hello!</p>");
+      cy.get(contentAnchor).should("have.text", "Hello!");
+
+      cy.log("Overwrite text");
+
+      cy.get(cesc("#\\/_codeeditor1") + " .cm-content").type(
+        "{control+a}<alert>Ovewritten!</alert>",
```

Comment:
Yeah, it looks like the save key combo is implemented to fire on control+s and command+s on any platform. So I assume on linux your windows or "super" key would also work for saving. The system native events won't do this, I just tested select all and I need to do command+a on mac, even in codemirror, same for undo. Redo on mac is command+shift+z, I didn't add it to the test in part to avoid adding another 5 second delay, but it's not like it is too big of a deal, and I guess I could amend the test to do something like 2 separate additions to the text, fire two undos and one redo and then just have a single wait and reload to test that those type of events will trigger a save.